### DISCUSSION TITLE
Understand and consolidate index data

### DISCUSSION THREAD

**Discussion by dee-corr** [milvus-io/milvus#21834] [Q&A and General discussion]
Hello,

I'm trying to consolidate my index files into one so I can visualise the clusters using feder https://github.com/zilliztech/feder

I'm struggling to understand the structure though. For example, this is a sample directory within the index directories:
/home/ec2-user/volumes/minio/a-bucket/files/index_files/438818504410696347/1/438818504362693112/438818504374493869

What do these ids mean?

The files contained in this dir:
BinaryIVF_0  BinaryIVF_1  BinaryIVF_10  BinaryIVF_11  BinaryIVF_12  BinaryIVF_2  BinaryIVF_3  BinaryIVF_4  BinaryIVF_5  BinaryIVF_6  BinaryIVF_7  BinaryIVF_8  BinaryIVF_9  indexParams  SLICE_META

Is it possible to consolidate index files into one to visualise with feder?

Any input would be greatly appreciated!!

**Comment by xiaofan-luan** [comment]
Hi dee-corr,
you actually need to merge or data into a single index.
But I guess @alwayslove2013 could offer more help

**Reply by xiaofan-luan** [reply]
If you want to look into where milvus handle small index files into a large one

segment_loader.go loadFieldIndexData is the function you want to look into

**Reply by xiaofan-luan** [reply]
the index meta I guess is the indexParams file, it should a json

**Reply by dee-corr** [reply]
Thank you @xiaofan-luan, I will take a look at that.

@alwayslove2013 do you have any additional tips that could help me?

Thanks!

**Reply by alwayslove2013** [reply]
Sorry for the limited index format supported by feder now. 

Visualizing data from milvus with feder is working in progress.

**Reply by dee-corr** [reply]
@alwayslove2013 thanks for your reply. By "work in progress" do you mean it won't be possible for me to visualise the indexes through Feder? Or do you mean that is it possible, but more difficult, as multiple index formats are not supported yet?

**Reply by alwayslove2013** [reply]
yeah, we are trying to make it possible~ 

we will let you know as soon as we achieve it.

**Reply by dee-corr** [reply]
Hi @alwayslove2013, wondering if you have any update on this?

**Reply by alwayslove2013** [reply]
Unfortunately, it's a little bit difficult and we haven't finished yet.
